web-common/src/features/canvas/stores/time-control.ts

```diff
@@ -0,0 +1,331 @@
+          allTimeRange.end,
+          selectedTimezone,
+        );
+
+        const timeRangeState = calculateTimeRangePartial(
+          allTimeRange,
+          selectedTimeRange,
+          undefined, // scrub not present in canvas yet
+          selectedTimezone,
+          defaultTimeRange,
+          minTimeGrain,
+        );
+        if (timeRangeState) return { ...timeRangeState };
+      },
+    );
+
+    this.comparisonRangeStateStore = derived(
+      [
+        specStore,
+        this.allTimeRange,
+        this.selectedComparisonTimeRange,
+        this.selectedTimezone,
+        this.showTimeComparison,
+        this.timeRangeStateStore,
+      ],
+      ([
+        spec,
+        allTimeRange,
+        selectedComparisonTimeRange,
+        selectedTimezone,
+        showTimeComparison,
+        timeRangeState,
+      ]) => {
+        if (!spec?.data || !timeRangeState) return undefined;
+        const timeRanges = spec.data?.canvas?.timeRanges;
+        return calculateComparisonTimeRangePartial(
+          timeRanges,
+          allTimeRange,
+          selectedComparisonTimeRange,
+          selectedTimezone,
+          undefined, // scrub not present in canvas yet
+          showTimeComparison,
+          timeRangeState,
+        );
+      },
+    );
+
+    this.setInitialState(specStore);
+  }
+
+  setInitialState(validSpecStore: CanvasSpecResponseStore) {
+    const selectedTimezone = get(this.selectedTimezone);
+    const comparisonTimeRange = get(this.selectedComparisonTimeRange);
+
+    const store = derived(
+      [this.allTimeRange, validSpecStore],
+      ([allTimeRange, validSpec]) => {
+        if (allTimeRange.isFetching) return;
+        const { defaultPreset } = validSpec.data?.canvas || {};
+        const timeRanges = validSpec?.data?.canvas?.timeRanges;
+
+        const defaultTimeRange = isoDurationToFullTimeRange(
+          defaultPreset?.timeRange,
+          allTimeRange.start,
+          allTimeRange.end,
+          selectedTimezone,
+        );
+
+        const newTimeRange: DashboardTimeControls = {
+          name: defaultTimeRange.name,
+          start: defaultTimeRange.start,
+          end: defaultTimeRange.end,
+        };
+
+        newTimeRange.interval = getTimeGrain(
+          undefined,
+          newTimeRange,
+          V1TimeGrain.TIME_GRAIN_UNSPECIFIED,
+        );
+
+        if (
+          defaultPreset?.comparisonMode ===
+          V1ExploreComparisonMode.EXPLORE_COMPARISON_MODE_TIME
+        ) {
+          const newComparisonRange = getComparisonTimeRange(
+            timeRanges,
+            allTimeRange,
+            newTimeRange,
+            comparisonTimeRange,
+          );
+          this.selectedComparisonTimeRange.set(newComparisonRange);
+          this.showTimeComparison.set(true);
+        }
+
+        this.selectedTimeRange.set(newTimeRange);
+      },
+    );
+
+    // Subscribe to ensure the derived code runs
+    store.subscribe(() => {});
```

Comment:
Dont we need to unsubscribe when the entity is removed/navigated from? Might lead to unintended callbacks.